### PR TITLE
Fix bug where pods would unintentionally share cgroupns

### LINKED ISSUE #6149: OCI runtime permission denied error running setns against proc
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

/kind feature

**Description**

Getting error

'''
Error: unable to start container 981e1f9359e97b99ba68e3fdceb3c80e887940999adf6e5bc6a06f944111a482: cannot setns `/proc/1525717/ns/cgroup`: Operation not permitted: OCI runtime permission denied error
'''

**Steps to reproduce the issue:**
1. podman pod create --name=family --share net
2. podman create --name=family_db_1 --pod=family -l io.podman.compose.config-hash=123 -l io.podman.compose.project=family -l io.podman.compose.version=0.0.1 -l com.docker.compose.container-number=1 -l com.docker.compose.service=db -e POSTGRES_PASSWORD=mypass --mount type=volume,source=family_db,destination=/var/lib/postgresql/data,exec --add-host nextcloud:127.0.0.1 --add-host family_nextcloud_1:127.0.0.1 --add-host db:127.0.0.1 --add-host family_db_1:127.0.0.1 postgres:12-alpine
3. podman start family_db_1

3.

**Describe the results you received:**
Error: unable to start container "f0ca1ddaa45f57115819ee995a946f6e2d446e872a71e08016fd00e3759cc66e": cannot setns `/proc/1532496/ns/cgroup`: Operation not permitted: OCI runtime permission denied error

**Describe the results you expected:**
should run container

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `podman version`:**

```
[joe@big-apple nextcloud (master)]$ podman --version
podman version 2.0.0-dev
```

**Output of `podman info --debug`:**

```
host:
  arch: amd64
  buildahVersion: 1.15.0-dev
  cgroupVersion: v2
  conmon:
    package: conmon-2.0.15-1.mga8
    path: /usr/bin/conmon
    version: 'conmon version 2.0.15, commit: c97b7dde0977ca31c535f1bcd5d6f720bf3bf0c7'
  cpus: 4
  distribution:
    distribution: mageia
    version: "8"
  eventLogger: file
  hostname: big-apple
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65636
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
  kernel: 5.6.10-server-1.mga8
  memFree: 34817998848
  memTotal: 67335356416
  ociRuntime:
    name: crun
    package: crun-0.13-1.mga8
    path: /usr/bin/crun
    version: |-
      crun version 0.13
      commit: e79e4de4ac16da0ce48777afb72c6241de870525
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +YAJL
  os: linux
  rootless: true
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-0.4.4-1.mga8
    version: |-
      slirp4netns version 0.4.4
      commit: d6d5ddec4d898c5592b273412c800bb384c9ac88
      libslirp: 4.2.0
  swapFree: 24847052800
  swapTotal: 24847052800
  uptime: 33h 57m 43.48s (Approximately 1.38 days)
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - registry.centos.org
  - docker.io
store:
  configFile: /home/joe/.config/containers/storage.conf
  containerStore:
    number: 4
    paused: 0
    running: 1
    stopped: 3
  graphDriverName: vfs
  graphOptions: {}
  graphRoot: /home/joe/.local/share/containers/storage
  graphStatus: {}
  imageStore:
    number: 5
  runRoot: /run/user/1000/containers
  volumePath: /home/joe/.local/share/containers/storage/volumes
version:
  Built: 1588962192
  GitCommit: 2d1291b406b2d00b177e5094073b9bab7d9fb5d7
  GoVersion: go1.14.2

```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
(paste your output here)
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

### PR DESCRIPTION
**Pull request by mheon**
This one was a massive pain to track down.

The original symptom was an error message from rootless Podman trying to make a container in a pod. I unfortunately did not look at the error message closely enough to realize that the namespace in question was the cgroup namespace (the reproducer pod was explicitly set to only share the network namespace), else this would have been quite a bit shorter.

I spent considerable effort trying to track down differences between the inspect output of the two containers, and when that failed I was forced to resort to diffing the OCI specs. That finally proved fruitful, and I was able to determine what should have been obvious all along: the container was joining the cgroup namespace of the infra container when it really ought not to have.

From there, I discovered a variable collision in pod config. The UsePodCgroup variable means "create a parent cgroup for the pod and join containers in the pod to it". Unfortunately, it is very similar to UsePodUTS, UsePodNet, etc, which mean "the pod shares this namespace", so an accessor was accidentally added for it that indicated the pod shared the cgroup namespace when it really did not. Once I realized that, it was a quick fix - add a bool to the pod's configuration to indicate whether the cgroup ns was shared (distinct from UsePodCgroup) and use that for the accessor.

Also included are fixes for `podman inspect` and `podman pod inspect` that fix them to actually display the state of the cgroup namespace (for container inspect) and what namespaces are shared (for pod inspect). Either of those would have made tracking this down considerably quicker.

Fixes #6149

### CODE REVIEW

**Comment by mheon** [comment]
Pushed one more commit with some small `podman inspect` tweaks for network mode

File: libpod/container_inspect.go [hunk 1]
```diff
@@ -698,6 +704,30 @@ func (c *Container) generateInspectContainerHostConfig(ctrSpec *spec.Spec, named
...
+					cgroupMode = fmt.Sprintf("ns:%s", ns.Path)
+				} else {
+					cgroupMode = "private"
+				}
+			}
+		}
+		if cgroupMode == "" {
+			cgroupMode = "host"
+		}
+	}
```
Comment (TomSweeneyRedHat) [inline libpod/container_inspect.go:728]: I thought I saw a similar function earlier today.  Could/should this be in an util.go kind of file?

**Comment by TomSweeneyRedHat** [comment]
One question for consideration.  Otherwise LGTM assuming happy tests.

**Comment by rhatdan** [comment]
LGTM

**Pull request merged**